src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int INITIAL_CAPACITY = 16;
+    public static final int GROW_VALUE = 2;
+    public static final float LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private Node<K, V>[] mapArray;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == 0) {
+            mapArray = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        }
+        if (putVal(key, value, mapArray)) {
+            if (++size > mapArray.length * LOAD_FACTOR) {
+                resize();
```

Comment:
if you only use resize with that if, better move it inside the method and change its name accordingly to reflect it may resize, or may not - sth like resizeIfNeeded()